chia/data_layer/data_store.py

```diff
@@ -81,6 +81,17 @@ async def create(cls, db_wrapper: DBWrapper) -> "DataStore":
                 )
                 """
             )
+            await self.db.execute(
+                """
+                CREATE TABLE IF NOT EXISTS ancestors(
+                    hash TEXT NOT NULL,
+                    ancestor TEXT,
+                    tree_id TEXT NOT NULL,
+                    generation INTEGER NOT NULL,
+                    PRIMARY KEY(hash, tree_id, generation)
+                )
+                """
+            )
```

Comment:
This does not work as is, the `root` foreign key complains.  I didn't actually try to fix it but I think that may be able to be addressed by just moving the root insertion earlier than the ancestor insertion.

```suggestion
            # TODO: Add ancestor -> hash relationship, this might involve temporarily
            #       deferring the foreign key enforcement due to the insertion order
            #       and the node table also enforcing a similar relationship in the
            #       other direction.
            #       FOREIGN KEY(ancestor) REFERENCES ancestors(ancestor)
            await self.db.execute(
                """
                CREATE TABLE IF NOT EXISTS ancestors(
                    hash TEXT NOT NULL REFERENCES node,
                    ancestor TEXT,
                    tree_id TEXT NOT NULL,
                    generation INTEGER NOT NULL,
                    PRIMARY KEY(hash, tree_id, generation),
                    FOREIGN KEY(tree_id, generation) REFERENCES root(tree_id, generation),
                    FOREIGN KEY(ancestor) REFERENCES node(hash)
                )
                """
            )
```